### PR TITLE
i.MX8MP: Add support for SPI 

### PR DESCRIPTION
**Pull request by leducp**
## Summary
Add support for SPI  (ecspi module)

## Testing
OK on Verdin board for SPI1

### CODE REVIEW

File: arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#define STATREG_TC                (1 << 7)
+#define STATREG_RO                (1 << 6)
+#define STATREG_RF                (1 << 5)
+#define STATREG_RDR               (1 << 4)
+#define STATREG_RR                (1 << 3)
+#define STATREG_TF                (1 << 2)
+#define STATREG_TDR               (1 << 1)
+#define STATREG_TE                (1 << 0)
+
+#define PERIODREG_CSD_CTL		  (16)
```
Comment (xiaoxiang781216) [inline arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h:90]: align

File: arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#define PERIODREG_CSRC			  (1 << 15)
+#define PERIODREG_SAMPLE_PERIOD	  (0)
+
+#define TESTREG_LBC               (1 << 31)
+#define TESTREG_RXCNT			  (8)
+#define TESTREG_TXCNT			  (0)
```
Comment (xiaoxiang781216) [inline arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h:96]: align

File: boards/arm/mx8mp/verdin-mx8mp/src/mx8mp_spi.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  return 0;
+}
+
+#ifdef CONFIG_SPI_CMDDATA
+int mx8mp_spi3_cmddata(struct spi_dev_s *dev, uint32_t devid, bool cmd)
+{
+  return -ENODEV;
+}
+#endif
+#endif
```
Comment (xiaoxiang781216) [inline boards/arm/mx8mp/verdin-mx8mp/src/mx8mp_spi.c:122]: add newline

File: arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h (hunk 1 above)
Comment (leducp) [reply arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h:90]: fixed

File: arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h (hunk 2 above)
Comment (leducp) [reply arch/arm/src/mx8mp/hardware/mx8mp_ecspi.h:96]: fixed

**Review by jerpelea — APPROVED** [review]

**Pull request merged**